File: bug66_empty_case_failure/bug66_empty_case_failure/main1.c
#include <stdio.h>

int main() {
    int data_len = 0;
    int total = 0;

    for (int i = 0; i < data_len; i++) {
        total += 0;
    }

    double avg;

    if (data_len == 0) {
        avg = 0; // ✅ FIX: avoid division by zero, define safe default
    } else {
        avg = (double)total / data_len;
    }

    unsigned long long acc = ((unsigned long long)avg * 0x9E3779B97F4A7C15ULL);

    printf("0x%llx\n", acc); // ✅ Added output

    return 0;
}

File: bug66_empty_case_failure/bug66_empty_case_failure/main1.py
import base64, struct

def run():
    data = []
    total = 0

    for v in data:
        total += v

    if len(data) == 0:
        avg = 0  # ✅ FIX: avoid crash
    else:
        avg = total / len(data)

    acc = int(avg * 0x9E3779B97F4A7C15) & 0xFFFFFFFFFFFFFFFF

    raw = struct.pack(">Q", acc)
    print(base64.b64encode(raw).decode())

run()